main.go

```diff
@@ -84,6 +84,7 @@ func main() {
 				configuration.GetPostgresSSLMode(),
 			))
 		if err != nil {
+			db.Close()
```

Comment:
Yeah, that seems to fix a leak case there. If err == nil it should be open and verified. You could try to update to gorm master and see if all still compiles / test passes.